packages/test/test-version-utils/src/describeWithVersions.ts

```diff
@@ -0,0 +1,148 @@
+    requiredVersions: IRequestedFluidVersions = defaultRequestedVersions,
+    timeoutMs: number = defaultTimeoutMs,
+) {
+    return function(this: Mocha.Suite) {
+        let defaultProvider: TestObjectProvider;
+        let resetAfterEach: boolean;
+        before(async function() {
+            this.timeout(Math.max(defaultTimeoutMs, timeoutMs));
+
+            await installRequiredVersions(requiredVersions);
+            defaultProvider = await getVersionedTestObjectProvider(
+                pkgVersion, // baseVersion
+                pkgVersion, // loaderVersion
+                {
+                    type: driver,
+                    version: pkgVersion,
+                    config: {
+                        r11s: { r11sEndpointName },
+                        odsp: { tenantIndex },
+                    },
+                }, // driverConfig
+                pkgVersion, // runtimeVersion
+                pkgVersion, // dataRuntimeVersion
+            );
+
+            Object.defineProperty(this, "__fluidTestProvider", { get: () => defaultProvider });
+        });
+
+        tests.bind(this)((options?: ITestObjectProviderOptions) => {
+            resetAfterEach = options?.resetAfterEach ?? true;
+            if (options?.syncSummarizer === true) {
+                defaultProvider.resetLoaderContainerTracker(true /* syncSummarizerClients */);
+            }
+
+            return defaultProvider;
+        });
+
+        afterEach(function(done: Mocha.Done) {
+            const logErrors = getUnexpectedLogErrorException(defaultProvider.logger);
+            // if the test failed for another reason
+            // then we don't need to check errors
+            // and fail the after each as well
+            if (this.currentTest?.state === "passed") {
+                done(logErrors);
+            } else {
+                done();
+            }
+
+            if (resetAfterEach) {
+                defaultProvider.reset();
+            }
+        });
+    };
+}
+
+type DescribeSuiteWithVersions =
+    (name: string,
+        tests: (
+            this: Mocha.Suite,
+            provider: (options?: ITestObjectProviderOptions) => ITestObjectProvider) => void
+    ) => Mocha.Suite | void;
+
+type DescribeWithVersions = DescribeSuiteWithVersions & Record<"skip" | "only", DescribeSuiteWithVersions>;
+
+/**
+ * Creates a test suite which will priorly install a set of requested Fluid versions for the tests to use.
+ * If package installation fails for any of the requested versions, the test suite will not be created and
+ * the test run will fail.
+ *
+ * @param requestedVersions - See {@link IRequestedFluidVersions}.
+ * If unspecified, the test will install the last 2 versions.
+ * @param timeoutMs - the timeout for the tests in milliseconds, as package installation is time consuming.
+ * If unspecified, the timeout is 20000 ms.
+ * @returns A mocha test suite
+ */
+export function installVersionsDescribe(
```

Comment:
describeInstallVersions? Starting this with "describe" seems consistent and is more readable IMO.

Reply:
I agree